### ISSUE TITLE
(Actions) Are action commands/sequences intended to be independent of each other?

### ISSUE THREAD

**Issue by mjzffr** [w3c/webdriver#1197]
Let's say the remote end receives two distinct action commands, one right after the other in the same session, that consist of a sequence of `pointerDown, pointerUp` on the same coordinates with a mouse pointer input source:

```
POST .../actions ... [pointerDown, pointerUp]
POST .../actions ... [pointerDown, pointerUp]
```

In effect, this generates two `click` events on the same element in quick succession, so my understanding is that a `dblclick` should also be generated. In other words, the first command impact the effect of the second in this case if they happen in a short enough interval.

Could anyone confirm that that's what is intended? Opinions? @ato @jgraham @whimboo

**Comment by whimboo** [comment]
I just checked the spec and was not able to find anything if action sequences should affect each other or if states should be cleared.

**Comment by jgraham** [comment]
If the spec doesn't say you clear the state then you  don't clear the state. Therefore I would expect the two commands to interact. *That said* I would find it surprising if two seperate commands shared timer state  (like click->double click) because that seems inherently racy. I would argue that the spec should be updated to make this clear.

**Comment by mjzffr** [comment]
> That said I would find it surprising if two seperate commands shared timer state (like click->double click) because that seems inherently racy.

I don't find it so surprising, since from the perspective of the document being interacted with, two clicks on the same element _did_ happen in a short interval -- why shouldn't that be a dblclick?

I think this could be addressed by the Release Actions command, which is supposed to reset the input state. ("It also clears all the internal state of the virtual devices.") It could include resetting timers. Doing it this way leaves it in the hands of the user to decide whether two commands interact or not.

**Comment by jgraham** [comment]
Because the result will depend on network latency, which seems pretty bad from the point of view of consistency. Sending |click| |click| might or might not produce a double click depending on external factors, whereas |click, click| always will (if you see what I mean).

**Comment by mjzffr** [comment]
Yeah, that's fair, but then again, anything with a pause or a timer is
susceptible to variability.
